core/controllers/profile_test.py

```diff
@@ -236,9 +238,16 @@ def test_get_profile_picture_valid_username(self):
         response_dict = self.get_json(
             '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
         )
-        # Although the user has a valid username, they have not yet supplied
-        # a profile picture.
-        self.assertIsNone(
+        self.assertIsNotNone(
+            response_dict['profile_picture_data_url_for_username'])
+
+    def test_get_signup_gravatar(self):
+        self.signup(self.EMAIL, self.USERNAME)
+        response_dict = self.get_json(
+            '%s%s' % (self.PROFILE_PIC_URL, self.USERNAME)
+        )
+        self.assertEqual(
+            self.get_gravatar_from_url(self.GRAVATAR_URL),
```

Comment:
That's not what I mean. The aim is to test the behaviour of fetch_gravatar(), __assuming__ that the urlfetch call works (or fails). The test should specify the result of the urlfetch call by swapping the call to urlfetch with a mock function that returns a deterministic value.